{{cookiecutter.repo_name}}/src/{{cookiecutter.repo_name_slug}}/{{cookiecutter.conversion_name}}/{{cookiecutter.conversion_name}}_convert_all_sessions.py

```diff
@@ -0,0 +1,120 @@
+"""Primary script to run to convert all sessions in a dataset using session_to_nwb."""
+from pathlib import Path
+from typing import Union
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pprint import pformat
+import traceback
+from tqdm import tqdm
+
+from .{{cookiecutter.conversion_name}}_convert_session import session_to_nwb
+
+
+def dataset_to_nwb(
+    *,
+    data_dir_path: Union[str, Path],
+    output_dir_path: Union[str, Path],
+    max_workers: int = 1,
+    stub_test: bool = False,
+    verbose: bool = True,
+):
+    """Convert the entire dataset to NWB.
+
+    Parameters
+    ----------
+    data_dir_path : Union[str, Path]
+        The path to the directory containing the raw data.
+    output_dir_path : Union[str, Path]
+        The path to the directory where the NWB files will be saved.
+    max_workers : int, optional
+        The number of workers to use for parallel processing, by default 1
+    stub_test : bool, optional
+        Whether to run a stub test, by default False
+    verbose : bool, optional
+        Whether to print verbose output, by default True
+    """
+    data_dir_path = Path(data_dir_path)
+    session_to_nwb_kwargs_per_session = get_session_to_nwb_kwargs_per_session(
```

Comment:
I guess this will be usually some form of path expansion? What else has gone here in your projects? Feel free to link I will link some examples.

Reply:
yeah, subject_id, start_date, and file_path are common.  I pass quite a few conversion options in as well in the lerner conversion: https://github.com/catalystneuro/lerner-lab-to-nwb/blob/654cb8bd59715e8d9901cd403ccd2d1b1f63b56c/src/lerner_lab_to_nwb/seiler_2024/seiler_2024_convert_session.py#L12

Reply:
But the main point is that it's flexible, each conversion should implement `get_session_to_nwb_kwargs_per_session` differently to match that conversion's specific needs.

Reply:
Sure, I understad that the point is the flexbility. I was asking for examples of implementations of `get_session_to_nwb_kwargs_per_session` the link that you shared is for one session. Maybe a mistake?

Reply:
Ah, I see.  I split it into two functions for the lerner lab -- here is one: https://github.com/catalystneuro/lerner-lab-to-nwb/blob/654cb8bd59715e8d9901cd403ccd2d1b1f63b56c/src/lerner_lab_to_nwb/seiler_2024/seiler_2024_convert_dataset.py#L107

Reply:
and here is the other: https://github.com/catalystneuro/lerner-lab-to-nwb/blob/654cb8bd59715e8d9901cd403ccd2d1b1f63b56c/src/lerner_lab_to_nwb/seiler_2024/seiler_2024_convert_dataset.py#L412

Reply:
Cool, thanks.